code/frontend/src/components/AddContractor/index.js

```diff
@@ -0,0 +1,244 @@
+      contractor[event.target.getAttribute('name')] = event.target.value;
+    } else {
+      const { projects } = state;
+      let dataIndex = parseInt(event.target.getAttribute('data-index'));
+      let project = projects[dataIndex];
+      project[event.target.getAttribute('name')] = event.target.value;
+      this.setState(Object.assign(state, {projects}));
+    }
+  }
+
+  handleDateInput(event) {
+    event.preventDefault();
+    const { state } = this;
+    const { projects } = state;
+    let dataIndex = parseInt(event.target.getAttribute('data-index'));
+    let project = projects[dataIndex];
+    project[event.target.getAttribute('name')] = event.target.value;
+    this.setState(Object.assign(state, {projects}));
+  }
+
+  handleRadioInput(event) {
+    const { state } = this;
+    if (!event.target.hasAttribute('data-index')) {
+      const { contractor } = state;
+      contractor[event.target.getAttribute('name')] = event.target.value;
+      this.setState(Object.assign(state, {contractor}));
+    } else {
+      const { projects } = state;
+      let dataIndex = event.target.getAttribute('data-index');
+      let project = projects[dataIndex];
+      project[event.target.getAttribute('name')] = event.target.value;
+      this.setState(Object.assign(state, {projects}));
+    }
+  }
+
+  handleAdd(event) {
+    event.preventDefault();
+    let projectState = this.state.projects;
+    projectState.push(this.createDefaultProjectObject());
+    this.setState({projects: projectState})
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { contractor } = this.state;
+    const { projects } = this.state;
+    this.props.onSubmit(contractor, projects, this.resetState);
+  }
+
+  createDefaultProjectObject() {
+    return {
+        projectName: '',
+        reportingmanagers: this.getReportingManagersOptions(),
+        hrpositions: this.getHrPositionOptions(),
+        ratetypes: this.getRateTypeOptions(),
+        hourlyrate: '',
+        paygrades: this.getPayGradeOptions(),
+        refnos: this.getRefNosOptions(),
+        mainSkills: this.getMainSkillsOptions(),
+        costCenters: this.getCostCenterOptions(),
+        originalDocumentation: '',
+        terminationNum: '',
+    };
+  }
+
+  getDropdownOptions (optionName) {
+    switch  (optionName) {
+      case "reportingManager":
+        return this.getReportingManagersOptions();
+      case "hrPosition":
+        return this.getHrPositionOptions();
+      case "rateType":
+        return this.getRateTypeOptions();
+      case "payGrade":
+        return this.getPayGradeOptions();
+      case "refNos":
+        return this.getRefNosOptions();
+      case "mainSkills":
+        return this.getMainSkillsOptions();
+      case "costCenters":
+        return this.getCostCenterOptions();
+      default:
+        return [];
+    }
+  }
+
+  getReportingManagersOptions() {
+    //fallbacks
+    return ['harry potter', 'luna luvgood'];
```

Comment:
Maybe remove these? I don't think hardcoded fallbacks like these would be good.
Same for the next few functions.

Reply:
made the changes for the single quotes but i kept the functions in case they are needed for api calls